rules/framework.bzl

```diff
@@ -410,7 +410,7 @@ def _get_direct_public_headers(provider, dep):
             return []
         return dep[provider].compilation_context.direct_public_headers
     elif provider == apple_common.Objc:
-        return dep[provider].direct_headers
+        return getattr(dep[provider], "direct_headers", [])
```

Comment:
@mattrobmattrob so the gist is that the previous condition is hit on Bazel 6. LGTM overall

Reply:
I didn't confirm that but as far as I can tell, yes. At least for all the cases tested by my code and the CI suite.